toolchain/semantics/semantics_handle_operator.cpp

```diff
@@ -51,7 +51,14 @@ auto SemanticsHandleInfixOperator(SemanticsContext& context,
               resume_block_id));
       return true;
     }
-
+    case TokenKind::Equal: {
+      // TODO: handle complex assignment expression such as `a = a + 1`.
+      context.ImplicitAsRequired(
+          parse_node, lhs_id, context.semantics_ir().GetNode(rhs_id).type_id());
+      context.AddNodeAndPush(parse_node, SemanticsNode::Assignment::Make(
+                                             parse_node, lhs_id, rhs_id));
```

Comment:
We should check that the LHS is something that can be assigned to. I'm adding a mechanism to check that in #3055, so for now please just add a TODO here.